packages/instrumentation-aws-sdk/src/services/dynamodb.ts

```diff
@@ -1,27 +1,45 @@
-import { SpanKind } from '@opentelemetry/api';
+import { Span, SpanKind, Tracer } from '@opentelemetry/api';
 import { RequestMetadata, ServiceExtension } from './ServiceExtension';
 import { SemanticAttributes } from '@opentelemetry/semantic-conventions';
-import { NormalizedRequest } from '../types';
+import { AwsSdkInstrumentationConfig, NormalizedRequest, NormalizedResponse } from '../types';
+import type { Request } from 'aws-sdk';
 
 export class DynamodbServiceExtension implements ServiceExtension {
-    requestPreSpanHook(request: NormalizedRequest): RequestMetadata {
+    requestPreSpanHook(normalizedRequest: NormalizedRequest): RequestMetadata {
         let spanKind: SpanKind = SpanKind.CLIENT;
         let spanName: string;
         let isIncoming = false;
-        const operation = request.commandName;
+        const operation = normalizedRequest.commandName;
 
         const spanAttributes = {
             [SemanticAttributes.DB_SYSTEM]: 'dynamodb',
-            [SemanticAttributes.DB_NAME]: request.commandInput?.TableName,
+            [SemanticAttributes.DB_NAME]: normalizedRequest.commandInput?.TableName,
             [SemanticAttributes.DB_OPERATION]: operation,
-            [SemanticAttributes.DB_STATEMENT]: JSON.stringify(request.commandInput),
+            [SemanticAttributes.DB_STATEMENT]: JSON.stringify(normalizedRequest.commandInput),
         };
 
+        if (operation == 'BatchGetItem') {
+            spanAttributes[SemanticAttributes.AWS_DYNAMODB_TABLE_NAMES] = Object.keys(
+                normalizedRequest.commandInput.RequestItems
+            );
+        }
+
         return {
             isIncoming,
             spanAttributes,
             spanKind,
             spanName,
         };
     }
+
+    responseHook(response: NormalizedResponse, span: Span, tracer: Tracer, config: AwsSdkInstrumentationConfig) {
+        const operation = response.request.commandName;
+
+        if (operation === 'BatchGetItem') {
+            span.setAttribute(
+                SemanticAttributes.AWS_DYNAMODB_CONSUMED_CAPACITY,
+                response.data.ConsumedCapacity.map((x: Object) => JSON.stringify(x))
```

Comment:
I'm not a typescript expert, but I usually see that `any` is used instead of `object`. Not sure what is best here, but we can also just use the right type for it: `ConsumedCapacity` (guess we need to `or` the type from v2 and v3 because this code serves both of them).
It's also fine the way it is right now, just taking the opportunity to discuss alternatives

Reply:
Regarding the `cannot read property 'map' of undefined`, I see in your tests that for this attribute to be populated, `ReturnConsumedCapacity` should be set on the request.
So to support the case of user performing the request without `ReturnConsumedCapacity`, we need to also check that the `ConsumedCapacity` property exists on the response (and a test for that can be amazing)

Reply:
> Do you know maybe when ConsumedCapacity is expected to be returned for a request? ... I see in your tests that for this attribute to be populated, ReturnConsumedCapacity should be set on the request.

Ah thank you so much for your diligent testing 🙂 You're absolutely right, I was really confused why the type was not showing up until I read on [the docs](https://docs.aws.amazon.com/AWSJavaScriptSDK/v3/latest/clients/client-dynamodb/interfaces/batchgetitemcommandinput.html#returnconsumedcapacity) that you have to add the optional `ReturnConsumedCapacity` field to get it to show up, but didn't go back and account for that in the instrumentation logic 😅 

I've added that check and test to go along with that! It requires duplicating the `mockAwsSend` but I guess it's unavoidable 😓 

> I usually see that any is used instead of object. Not sure what is best here, but we can also just use the right type for it: ConsumedCapacity

This is a great prompt! I dug into the deep recollections of my TS knowledge from years ago and came up with this 😄 

```node
import type { ConsumedCapacity as ConsumedCapacityV2 } from 'aws-sdk/clients/dynamodb';
import type { ConsumedCapacity as ConsumedCapacityV3 } from '@aws-sdk/client-dynamodb';

type ConsumedCapacity = ConsumedCapacityV2 | ConsumedCapacityV3;
```

I had to add `@aws-sdk/client-dynamodb` to the `dev-dependencies` but since we already do that for S3 and SQS I guessed it was fine 🙂